[Vue] migrate popover handler to popover service

diosmosis opened the pull request:
### Description:

**This PR depends on #18443.**

Changes:
* The current popover handler directive which if placed on a page handles the popover parameter has been replaced with a service class that does the same thing but doesn't need any HTML placed on the page.

![image](https://user-images.githubusercontent.com/125140/144876381-21aa3914-d302-4bb8-8872-41de95b98ee8.png)

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

diosmosis committed:
migrate widget bydimension container

diosmosis committed:
migrate widget + add tooltips directive

diosmosis committed:
quick fix

justinvelluppillai committed:
Updating version to 4.6.0

diosmosis committed:
loading in page

diosmosis committed:
update expected screenshot

diosmosis committed:
Merge branch 'vue-period-selector' into vue-reporting-pages-model

diosmosis committed:
add wait just in case travis is slow

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
fix ordering bug

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
add another wait

diosmosis committed:
Merge branch '4.x-dev' into vue-reporting-menu

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch '4.x-dev' into vue-reporting-menu

diosmosis committed:
css tweak

diosmosis committed:
fix some bugs and tests

diosmosis committed:
undo screenshot changes

diosmosis committed:
Menus test passing locally

diosmosis committed:
[Vue] date picker viewDate property is not kept up to date (#18385)

* viewDate ref is not kept up to date

* rebuild corehome

diosmosis committed:
reporting menu subcategory items are meant to be normal links

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
update some screenshots

diosmosis committed:
use innerText instead of text() since angularjs maintains newlines in HTML that vue does not add

diosmosis committed:
trigger angularjs digest after ajaxhelper request

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
update screenshots, fix bug in link generation in reporting menu and allow syncing multiple screenshot regexes at a time

diosmosis committed:
undo box-shadow change for UI tests

diosmosis committed:
fix more issues & update more tests

diosmosis committed:
update some screenshots

diosmosis committed:
fix some tests

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
rebuild CoreHome

justinvelluppillai committed:
Merged 4.x-dev

diosmosis committed:
quick fix

diosmosis committed:
Merge branch '4.x-dev' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

justinvelluppillai committed:
built vue files

diosmosis committed:
fix angularjs issue

diosmosis committed:
add comment

diosmosis committed:
update umd files

diosmosis committed:
Merge branch 'next_release' into vue-widget-loader

diosmosis committed:
Merge branch 'next_release' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

mattab committed:
4.6.1-rc1

mattab committed:
4.6.1

diosmosis committed:
emit/broadcast on correct scope in wrapper

diosmosis committed:
Merge branch '4.x-dev' into vue-form-field

diosmosis committed:
fix field array title

diosmosis committed:
apply some pr feedback

diosmosis committed:
apply more pr feedback

diosmosis committed:
another fix

diosmosis committed:
tweak

diosmosis committed:
fix ng-change not executed before ng-model

diosmosis committed:
fix another set of issues

diosmosis committed:
fix another issue

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch '4.x-dev' into vue-form-field

diosmosis committed:
Merge branch 'vue-form-field' into vue-reporting-menu

diosmosis committed:
better ng-change/ng-model fix

diosmosis committed:
Merge branch 'vue-form-field' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-form-field' into vue-period-selector

diosmosis committed:
update some screenshots

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-period-selector' into vue-reporting-pages-model

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
remove some TODOs

diosmosis committed:
initiate initial ng-change ONLY for site selectors where this behavior applies

diosmosis committed:
Merge branch 'vue-form-field' into vue-period-selector

diosmosis committed:
Merge branch 'vue-period-selector' into vue-reporting-pages-model

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch '4.x-dev' into vue-period-selector

diosmosis committed:
Merge branch 'vue-period-selector' into vue-reporting-pages-model

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
Merge branch 'next_release' into vue-widget-loader

diosmosis committed:
Merge branch 'vue-post-event-regression' into vue-widget-loader

diosmosis committed:
fix some issues

diosmosis committed:
Merge branch 'vue-widget-loader' into vue-widget

diosmosis committed:
couple more fixes

diosmosis committed:
fix another title issue

diosmosis committed:
Merge branch 'vue-widget-loader' into vue-widget

diosmosis committed:
rebuild vue

diosmosis committed:
do not report on ajax errors in notifications if not logged in

diosmosis committed:
migrate reporting page and model

diosmosis committed:
rebuild vue

diosmosis committed:
create sites selector model adapter

diosmosis committed:
fix siteselector vue bug, initial site is only set if there is just one site available

diosmosis committed:
rebuild vue

diosmosis committed:
migrate plugin settings directive

diosmosis committed:
remove TODO

diosmosis committed:
migrate plugin filter directive

diosmosis committed:
migrate two more plugins directives

diosmosis committed:
migrate save button

diosmosis committed:
fix a bunch of bugs

diosmosis committed:
Merge branch 'vue-widget' into vue-core-plugins-admin

diosmosis committed:
fix another widget bug

diosmosis committed:
Merge branch 'vue-widget' into vue-core-plugins-admin

diosmosis committed:
allow change event name between angularjs and vue

diosmosis committed:
rebuild vue

diosmosis committed:
migrate plugin form directive

diosmosis committed:
get to work

diosmosis committed:
migrate select-on-focus directive and start migrating report-export directive

diosmosis committed:
finish migrating report export directive & popover component + create reusable function to create vue app and add globals to it

diosmosis committed:
rebuild vue

diosmosis committed:
remove angularjs files and move less contents to vue dir

diosmosis committed:
migrate sparkline component

diosmosis committed:
migrate progressbar component

diosmosis committed:
migrate content intro & content table directives

diosmosis committed:
rebuild vue

diosmosis committed:
migrate piwik-ajax-form component to vue

diosmosis committed:
migrate popover handler directive to service class

diosmosis committed:
Merge branch '4.x-dev' into vue-reporting-pages-model

diosmosis committed:
built vue files

diosmosis committed:
fix function signature

diosmosis committed:
fix vue warning

diosmosis committed:
fix ajax request race condition

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
add new notification type "help" so the help notification is not cleared when clearing transient notifications

diosmosis committed:
fix some bugs and tests

diosmosis committed:
update screenshot

diosmosis committed:
Merge branch 'vue-reporting-pages-model' into vue-reporting-menu

diosmosis committed:
update screenshot & fix a test

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
allow using unminified jquery ui + fix bug in last fix

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
fix error when enrichedheadline is used in modal

diosmosis committed:
add polyfill min.js

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
remove two todos

diosmosis committed:
fix widget url logic

diosmosis committed:
update some screenshots and fix sanitization/escape issue

diosmosis committed:
update screenshots

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
Merge branch 'vue-widget-loader' into vue-widget

diosmosis committed:
fix url location updating regression in MatomoUrl.updateLocation use

diosmosis committed:
submodule

diosmosis committed:
Merge branch 'vue-period-selector-regression' into vue-widget

diosmosis committed:
update screenshots and fix possible error in json parse

diosmosis committed:
Merge branch 'vue-period-selector-regression' into vue-widget-loader

diosmosis committed:
built vue files

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
Merge branch 'vue-period-selector-regression' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-period-selector-regression' into vue-reporting-menu

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
use correct variable

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-period-selector-regression' into vue-reporting-menu

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
fix widget url logic

diosmosis committed:
segment parameter can be undefined now for some reason

diosmosis committed:
fix ngmodel binding in siteselector adapter (for last time hopefully)

diosmosis committed:
the original site selector only set the first site to the first site in the initial sites query if there was only one site in the entire matomo instance

diosmosis committed:
fix sitesmanager ui test failure

diosmosis committed:
fix usersettings test failure

diosmosis committed:
rebuild vue

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
more siteselector tweaks.

diosmosis committed:
more siteselector tweaks.

diosmosis committed:
build CoreHome

diosmosis committed:
another siteselector issue

diosmosis committed:
update screenshots

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
Merge branch 'vue-widget-loader' into vue-widget

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
update screenshot and try to fix random failure

diosmosis committed:
Merge branch 'vue-reporting-menu' into vue-widget-loader

diosmosis committed:
fix some issues in widget.vue when containerid is specified

diosmosis committed:
fix couple tests

diosmosis committed:
Merge branch 'vue-widget-loader' into vue-widget

diosmosis committed:
fix several test failures

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
Merge branch 'vue-reporting-page' into vue-site-selector-model

diosmosis committed:
Merge branch 'vue-site-selector-model' into vue-plugin-settings

diosmosis committed:
fix string concat

diosmosis committed:
fix test failure

diosmosis committed:
extra change

diosmosis committed:
fix last change and random failure

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
styling fix

diosmosis committed:
fix last fix

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
real fix this time

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
fix stray request

diosmosis committed:
Merge branch 'vue-widget' into vue-reporting-page

diosmosis committed:
proper fix

diosmosis committed:
update build files

diosmosis committed:
Merge branch 'vue-reporting-page' into vue-site-selector-model

diosmosis committed:
Merge branch 'vue-site-selector-model' into vue-plugin-settings

diosmosis committed:
try to fix random failure

diosmosis committed:
Merge branch 'vue-reporting-page' into vue-plugin-settings

diosmosis committed:
do not submit form

diosmosis committed:
check for api errors in promise chain in ajaxhelper.ts

diosmosis committed:
force a digest after a location change

diosmosis committed:
Merge branch 'vue-plugin-settings' into vue-core-plugins-admin

diosmosis committed:
Merge branch 'vue-core-plugins-admin' into vue-report-export

diosmosis committed:
Merge branch 'vue-report-export' into vue-corehome-components-1

diosmosis committed:
use proper abortcontroller method instead of promise hack, have to add new polyfill + try to fix random test failure

diosmosis committed:
Merge branch 'vue-plugin-settings' into vue-core-plugins-admin

diosmosis committed:
some UI test fixes

diosmosis committed:
Merge branch 'vue-core-plugins-admin' into vue-report-export

diosmosis committed:
fix some report export issues

diosmosis committed:
several save button fixes + make replace approximation in createAngularJsAdapter better

diosmosis committed:
Merge branch 'vue-core-plugins-admin' into vue-report-export

diosmosis committed:
Merge branch 'vue-report-export' into vue-corehome-components-1

diosmosis committed:
fix sparkline

diosmosis committed:
apply after manual click triggering in savebutton

diosmosis committed:
Merge branch 'vue-core-plugins-admin' into vue-report-export

diosmosis committed:
Merge branch 'vue-report-export' into vue-corehome-components-1

diosmosis committed:
css fixes and piwik-content-table was never applied by angularjs in installtion

diosmosis committed:
rebuild vue

diosmosis committed:
add names to divs so they can still be queried as they were in angularjs

diosmosis committed:
rebuild vue

diosmosis committed:
update expected screenshots

diosmosis committed:
Merge branch 'vue-corehome-components-1' into vue-ajax-form

diosmosis committed:
Merge branch 'vue-ajax-form' into vue-popover-handler

diosmosis committed:
rebuild vue

diosmosis committed:
now that format_metrics checkbox works, need to check it

diosmosis committed:
Merge branch 'vue-report-export' into vue-corehome-components-1

diosmosis committed:
Merge branch 'vue-corehome-components-1' into vue-ajax-form

diosmosis committed:
Merge branch 'vue-ajax-form' into vue-popover-handler

diosmosis committed:
small delay before processing first popover

diosmosis committed:
fix overlay test

diosmosis commented:
UI tests should be passing for this one

peterhashair commented:
maybe we need to resolve the conflicts, I am trying to fix those local, but it seems to return a blank page without any javascript errors on my tests.

diosmosis committed:
Merge branch '4.x-dev' into vue-corehome-components-1

diosmosis committed:
Merge branch 'vue-corehome-components-1' into vue-ajax-form

diosmosis committed:
remove unintended changes

diosmosis committed:
remove unintended changes

diosmosis committed:
Merge branch 'vue-ajax-form' into vue-popover-handler

bx80 committed:
Show a summary of new features (#18065)

* Added "What is new" notification display, populated by a new event

* Removed test example event hook

* Added support for applying a link attribute to menu items, fixes layout issue for mobile with html menu items

* Updated UI test screenshots

* Revert accidental edit

* Hide the "What's new" icon if there are no new features to show

* Changed to use changes.json, track user last viewed, added ui test

* Fix UserManager unit tests broken by new ts_changes_viewed user field

* Moved getChanges to separate helper class, added unit test, added user view access check

* Updated to add new changes table and populate only on plugin update/install

* Added missing fixture class, updated UI screenshots

* Updated matomo font to add ringing bell and new releases icons

* Fix for integration test

* Reworked class structure, removed unnecessary angular directive, merged templates, other tidy ups

* built vue files

* built vue files

* Added null user check, missing table exception handling, show plugin name in change title, better handling of missing change fields

* Added sample changes file, moved UserChanges db code to changes model, added return type hints, better db error code handling, various other improvements

* Revert accidental UI screenshot commit

* Fix for incorrect link name parameter in sample changes, switched back to using $db->query for INSERT IGNORE

* Integration test fix, UI screenshot updates

* Test fix

* Added link styling, show CoreHome changes without plugin prefix in title

* Update UI test screenshot

* Added styles to the popover, added event for filtering changes

* Test fix

* UI test screenshot updates

Co-authored-by: sgiehl <stefan@matomo.org>
Co-authored-by: bx80 <bx80@users.noreply.github.com>

Commit:
Update test translation (#18531)

update a test failed XML

github-actions[bot] committed:
updates all submodules (#18541)

Co-authored-by: diosmosis <diosmosis@users.noreply.github.com>

weblate committed:
Translations update from Hosted Weblate (#18529)

* Translated using Weblate (Greek)

Currently translated at 100.0% (162 of 162 strings)

Translation: Matomo/Plugin CoreAdminHome
Translate-URL: https://hosted.weblate.org/projects/matomo/plugin-coreadminhome/el/

[ci skip]

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Vasilis Lourdas <dev@lourdas.eu>

* Translated using Weblate (Chinese (Simplified))

Currently translated at 83.9% (136 of 162 strings)

Translation: Matomo/Plugin CoreAdminHome
Translate-URL: https://hosted.weblate.org/projects/matomo/plugin-coreadminhome/zh_Hans/

[ci skip]

Translated using Weblate (Chinese (Simplified))

Currently translated at 99.6% (620 of 622 strings)

Translation: Matomo/Matomo Base
Translate-URL: https://hosted.weblate.org/projects/matomo/matomo-base/zh_Hans/

[ci skip]

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: 刘韬 <lyuutau@outlook.com>

* Update translation files

Updated by "Squash Git commits" hook in Weblate.

Translation: Matomo/Plugin CoreAdminHome
Translate-URL: https://hosted.weblate.org/projects/matomo/plugin-coreadminhome/

[ci skip]

Co-authored-by: Vasilis Lourdas <dev@lourdas.eu>
Co-authored-by: 刘韬 <lyuutau@outlook.com>

diosmosis committed:
[Vue] migrate report export directive and popover (#18440)

* update files

* sidenav start

* make getRef a utility method

* tweak

* add return type

* finish converting side-nav directive

* starting on reporting menu conversion

* remove unused properties

* convert reporting pages service

* migrate report metadata store

* remove angularjs files

* migrating reporting pages store

* make store adapters more immutable

* get service adapters to work

* fix a UI test

* another html fix

* migrate most of reporting menu directive and model

* Use themed font family for input forms to override materialize.css styling

* rebuild vue

* add a missing div

* ui test fixes

* update styling

* get to build

* get to load in the UI w/o error

* clone result of functions

* fix compile issue

* migrate widget loader and get to load in UI

* rebuild vue

* migrate widgetcontainer

* migrate widget bydimension container

* migrate widget + add tooltips directive

* quick fix

* Updating version to 4.6.0

* loading in page

* update expected screenshot

* add wait just in case travis is slow

* fix ordering bug

* add another wait

* rebuild vue

* css tweak

* fix some bugs and tests

* undo screenshot changes

* Menus test passing locally

* [Vue] date picker viewDate property is not kept up to date (#18385)

* viewDate ref is not kept up to date

* rebuild corehome

* reporting menu subcategory items are meant to be normal links

* update some screenshots

* use innerText instead of text() since angularjs maintains newlines in HTML that vue does not add

* trigger angularjs digest after ajaxhelper request

* rebuild vue

* update screenshots, fix bug in link generation in reporting menu and allow syncing multiple screenshot regexes at a time

* undo box-shadow change for UI tests

* fix more issues & update more tests

* update some screenshots

* fix some tests

* rebuild CoreHome

* quick fix

* built vue files

* fix angularjs issue

* add comment

* update umd files

* 4.6.1-rc1

* 4.6.1

* fix field array title

* apply some pr feedback

* apply more pr feedback

* another fix

* tweak

* fix ng-change not executed before ng-model

* fix another set of issues

* fix another issue

* rebuild vue

* better ng-change/ng-model fix

* update some screenshots

* rebuild vue

* remove some TODOs

* initiate initial ng-change ONLY for site selectors where this behavior applies

* emit/broadcast on correct scope in wrapper

* rebuild vue

* fix some issues

* couple more fixes

* fix another title issue

* rebuild vue

* do not report on ajax errors in notifications if not logged in

* migrate reporting page and model

* rebuild vue

* create sites selector model adapter

* fix siteselector vue bug, initial site is only set if there is just one site available

* rebuild vue

* migrate plugin settings directive

* remove TODO

* migrate plugin filter directive

* migrate two more plugins directives

* migrate save button

* fix a bunch of bugs

* fix another widget bug

* allow change event name between angularjs and vue

* rebuild vue

* migrate plugin form directive

* get to work

* migrate select-on-focus directive and start migrating report-export directive

* finish migrating report export directive & popover component + create reusable function to create vue app and add globals to it

* rebuild vue

* remove angularjs files and move less contents to vue dir

* built vue files

* fix function signature

* fix vue warning

* fix ajax request race condition

* rebuild vue

* add new notification type "help" so the help notification is not cleared when clearing transient notifications

* fix some bugs and tests

* update screenshot

* update screenshot & fix a test

* allow using unminified jquery ui + fix bug in last fix

* fix error when enrichedheadline is used in modal

* add polyfill min.js

* remove two todos

* fix widget url logic

* update some screenshots and fix sanitization/escape issue

* update screenshots

* rebuild vue

* fix url location updating regression in MatomoUrl.updateLocation use

* submodule

* update screenshots and fix possible error in json parse

* built vue files

* Merge branch 'vue-period-selector-regression' into vue-reporting-menu

* rebuild vue

* use correct variable

* rebuild vue

* fix widget url logic

* segment parameter can be undefined now for some reason

* fix ngmodel binding in siteselector adapter (for last time hopefully)

* the original site selector only set the first site to the first site in the initial sites query if there was only one site in the entire matomo instance

* fix sitesmanager ui test failure

* fix usersettings test failure

* rebuild vue

* more siteselector tweaks.

* build CoreHome

* more siteselector tweaks.

* another siteselector issue

* update screenshots

* update screenshot and try to fix random failure

* fix some issues in widget.vue when containerid is specified

* fix couple tests

* fix several test failures

* fix string concat

* fix test failure

* extra change

* fix last change and random failure

* styling fix

* fix last fix

* real fix this time

* fix stray request

* proper fix

* update build files

* try to fix random failure

* do not submit form

* check for api errors in promise chain in ajaxhelper.ts

* force a digest after a location change

* use proper abortcontroller method instead of promise hack, have to add new polyfill + try to fix random test failure

* some UI test fixes

* fix some report export issues

* several save button fixes + make replace approximation in createAngularJsAdapter better

* apply after manual click triggering in savebutton

* add names to divs so they can still be queried as they were in angularjs

* rebuild vue

* now that format_metrics checkbox works, need to check it

* fix unintended changes

* updated expected screenshots

* update two more

* go back to previous format_metrics behavior in popover

Co-authored-by: Justin Velluppillai <justin@innocraft.com>
Co-authored-by: justinvelluppillai <justinvelluppillai@users.noreply.github.com>
Co-authored-by: Matthieu Aubry <mattab@users.noreply.github.com>

diosmosis committed:
Merge branch '4.x-dev' into vue-corehome-components-1

peterhashair commented on plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue:
just wondering, will $emit function works here.

peterhashair reviewed (commented):
just one suggestion, if that works

diosmosis commented on plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue:
I'm not sure what you mean, how would $emit be used here?

diosmosis committed:
[Vue] remove support in vue for FormField.allSettings (#18542)

* deprecate support in vue for FormField.allSettings since deep watching the property doesnt quite work

* built vue files

diosmosis committed:
Merge branch '4.x-dev' into vue-corehome-components-1

diosmosis committed:
update screenshots

diosmosis committed:
update screenshots

diosmosis committed:
Merge branch 'vue-corehome-components-1' into vue-ajax-form

peterhashair commented on plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue:
ah, actually i confused myself, I was thinking this 
```html
 <div ref="root" @click=‘this.submitForm()’>
```

peterhashair commented on plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue:
ah, actually I confused myself, I see why this way.

diosmosis committed:
update screenshot

diosmosis committed:
Merge branch 'vue-corehome-components-1' into vue-ajax-form

diosmosis committed:
Merge branch 'vue-ajax-form' into vue-popover-handler

diosmosis committed:
Merge branch '4.x-dev' into vue-popover-handler
